app/rest/metrics-collector/src/main/java/io/syndesis/rest/metrics/collector/MetricsCollector.java

```diff
@@ -0,0 +1,318 @@
+                .filter(p -> Readiness.isReady(p))
+                .forEach(p ->
+                    executor.execute(new PodMetricsReader(
+                            kubernetes,
+                            p.getMetadata().getName(),
+                            p.getMetadata().getLabels().get("integration"),
+                            p.getMetadata().getLabels().get("syndesis.io/integration-id"),
+                            p.getMetadata().getLabels().get("syndesis.io/deployment-id"),
+                            rawMetricsHandler))
+            );
+
+            ListResult<Integration> integrationList = dataManager.fetchAll(Integration.class);
+            Set<String> activeIntegrationIds = new HashSet<>();
+            for (Integration integration : integrationList.getItems()) {
+                String name = integration.getName();
+                String integrationId = integration.getId().get();
+                LOGGER.info("IntegrationId=" + integrationId);
+                activeIntegrationIds.add(name);
+
+                Map<String,RawMetrics> rawMetrics = getRawMetrics(name);
+                IntegrationMetricsSummary currentSummary =
+                        computeIntegrationSummary(integrationId, rawMetrics, livePods);
+                IntegrationMetricsSummary existingSummary =
+                        dataManager.fetch(IntegrationMetricsSummary.class, integrationId);
+                if (existingSummary == null) {
+                    dataManager.create(currentSummary);
+                } else if (existingSummary.hashCode() != currentSummary.hashCode()) {
+                    //only write to the DB when the new metrics differs to unnecessary
+                    //and expensive writes to the DB
+                    dataManager.update(currentSummary);
+                }
+                curateDeadPodMetrics(integrationId, name, rawMetrics, livePods);
+            }
+            curateDeletedIntegrationMetrics(activeIntegrationIds);
+
+        } catch (@SuppressWarnings("PMD.AvoidCatchingGenericException") IOException ex) {
+            LOGGER.error("Error while iterating integration pods.", ex);
+        }
+
+    }
+
+    /**
+     * Obtains all RawMetrics entries in the DB for the current integration
+     *
+     * @param integrationId - the integrationId for which we are obtaining the metrics
+     * @return a Map containing all RawMetrics entries for the current integration,
+     * the key is either HISTORY or the podName.
+     * @throws IOException
+     */
+    /*default*/ Map<String,RawMetrics> getRawMetrics(String integrationId) throws IOException {
+        //try to obtain metrics in this integration
+        Map<String,RawMetrics> metrics = new HashMap<>();
+        String json = jsonDB.getAsString(path(integrationId));
+        LOGGER.debug("JSON: " + json);
+        if (json != null) {
+            metrics = Json.mapper().readValue(json, new TypeReference<Map<String,RawMetrics>>() {});
```

Comment:
A DAO method with proper serialization would be useful here, as we do this for any other objects., too.